### ISSUE TITLE
From -1 to 1 or from 0 to 1 ?

### ISSUE THREAD

**Issue by Cristianasp** [zygmuntz/gaussrank#2]
Hello !!

I read your article and it says:
"First step is to assign a linspace to the sorted features from -1…1, then apply the inverse of error function ErfInv to shape them like gaussians, then I substract the mean. Binary features are not touched with this trafo (eg. 1-hot ones). This works usually much better than standard mean/std scaler or min/max."
http://fastml.com/preparing-continuous-features-for-neural-networks-with-rankgauss/

But you mentioned Jahrer interview and he says something different:
"normalization
Input normalization for gradient-based models such as neural nets is critical. For lightgbm/xgb it does not matter. The best what I found during the past and works straight of the box is "RankGauss". Its based on rank transformation. First step is to assign a linspace to the sorted features from 0..1, then apply the inverse of error function ErfInv to shape them like gaussians, then I substract the mean. Binary features are not touched with this trafo (eg. 1-hot ones). This works usually much better than standard mean/std scaler or min/max."
https://www.kaggle.com/c/porto-seguro-safe-driver-prediction/discussion/44629

So should it be sort from -1 to 1 or sort from 0 to 1 ?

Thanks

**Comment by zygmuntz** [comment]
It's the same snippet, looks like he edited it. I would guess it doesn't make that much difference, with 0...1 after subtracting the mean you'll get a similar gaussian with a mean of zero as you'd get from -1...1. I may be wrong though.
